server/server.go

```diff
@@ -662,9 +675,18 @@ func (s *server) CreateSpecificBlockchains(
 	// Create chains response
 	chainIDs := make([]string, len(chains))
 	for i, chain := range chains {
-		chainIDs[i] = chain.BlockchainID.String()
+		// Copy string to avoid any shared pointers with data in [ClusterInfo]
+		chainIDs[i] = strings.Clone(chain.BlockchainID.String())
+	}
+
+	// Deep copy struct to avoid shared pointers in
+	// gRPC marshaling (which can trigger race conditions and message decoding
+	// failure)
+	clusterInfo, err := deepCopy(s.clusterInfo)
+	if err != nil {
+		return nil, err
 	}
-	return &rpcpb.CreateSpecificBlockchainsResponse{ClusterInfo: s.clusterInfo, Chains: chainIDs}, nil
+	return &rpcpb.CreateSpecificBlockchainsResponse{ClusterInfo: clusterInfo, Chains: chainIDs}, nil
```

Comment:
Technically, returning `s.clusterInfo` in this response is not safe as it may be modified as soon as the lock is released.